Automate icingaweb2 setup

arjanoosting opened the pull request:
As discussed the code to automate the icingaweb2 setup (Fixes #2)
This is still work in progress...

I introduced quite a lot of new variables and I am not really sure about the naming convention for those yet. 

TODO: Also not all the _default and _custom variants are there yet

Also for Icingaweb you always? wan't an Icinga API user. So we maybe might just as well create that one in the default playbook when i2_setup_webui is set?

TODO: The OS user and group of the IcingaWeb configuration should have defaults based on the OS.

I am still figuring out how the other IcingaWeb2 modules need to be installed, probably I also will change some stuff there as well as we will be using the Director and GraphiteWeb modules

arjanoosting committed:
Add missing dict2items filter

arjanoosting committed:
Also install php-curl for API access and php-imagick for reports

arjanoosting committed:
Make sure apache is restarted when new PHP modules are installed/enabled

arjanoosting committed:
Generate configuration files for IcingaWeb2 based on variables

arjanoosting committed:
Merge branch 'development' into automate-icingaweb2-setup

kwisatz commented:
Thanks, I'll take a look at it. Diffs of this size are easier to digest…

kwisatz commented on tasks/icinga2-web2.yml:
```diff
@@ -5,6 +5,44 @@
+  become: true
+  template:
+    src: "icingaweb2-config.ini.j2"
+    dest: "/{{ item.dest }}"
```
Your `dest` below are already absolute.
Now, I might have used `dest: "/etc/icingaweb2/{{ item.dest }}"` here, but I can also see the advantage of having the absolute path below. I'd still refactor this and move the common root up here.

kwisatz reviewed (commented):
Looks good! I'd just wait for you to act on your #TODO items, such as the `i2_webui_user`.

I'd perhaps just use a regular variable for the api user password now and do the vault stuff in a separate diff/PR?

Also, we already have `i2_api_users` on the masters (in my case, inside `group_vars/masters/basic.yml`), shouldn't we try to re-use them? 
Well, either that, or as you said, have the config here, but then automatically generate the api user. So that'd mean, e.g. before running the ` Master - Create Api-Users config file` task, merge this user into `i2_api_users` ?!

arjanoosting committed:
Quote zone names

arjanoosting committed:
Merge branch 'development' into automate-icingaweb2-setup

arjanoosting committed:
Use the i2_api_host and i2_api_port to connect to parent

arjanoosting committed:
Merge branch 'development' into automate-icingaweb2-setup

arjanoosting committed:
Adjust the way we handle the configuration files so we can reuse it for
the additional modules as well

arjanoosting committed:
Adjust the installation of modules to include an version and repo
Handle additional actions for director module

arjanoosting committed:
Fixes in the node hierachy detection

arjanoosting committed:
Merge branch 'development' into automate-icingaweb2-setup

arjanoosting committed:
Fix name of default config and order options in generated files

arjanoosting committed:
Use template instead of blockinfile for the api-users.conf file

arjanoosting committed:
Cleanup unused code

arjanoosting committed:
use openssl_certificate_info

arjanoosting committed:
Merge branch 'development' into automate-icingaweb2-setup

arjanoosting committed:
Install icinga-director service, create Director database and create all configuration directories

arjanoosting committed:
Simple label for loop and set managed_object_files without loop

arjanoosting committed:
Merge branch 'development' into automate-icingaweb2-setup

arjanoosting committed:
Setup the icinga nodes in layers, first masters, then satellites, then
the agents

arjanoosting committed:
Merge branch 'development' into automate-icingaweb2-setup

arjanoosting committed:
Fix the layout of the feature configuration, which fixes IDO cleanup
configuration generation

arjanoosting committed:
Merge branch 'development' into automate-icingaweb2-setup

arjanoosting committed:
Fix layout of generated configuration

arjanoosting committed:
Merge branch 'development' into automate-icingaweb2-setup

arjanoosting committed:
Configure the satellites via "Top Down Config Sync"

arjanoosting committed:
Merge branch 'development' into automate-icingaweb2-setup

arjanoosting committed:
Mark commands tags which don't change the system

arjanoosting committed:
Make the state of the icinga packages a variable so we can use this role
to upgrade the packages as well

arjanoosting committed:
Merge branch 'development' into automate-icingaweb2-setup

arjanoosting committed:
Use the i2_package_state for Red Hat as well and use the restart icinga2
handler so icinga2 will be restarted on upgrades

arjanoosting committed:
Merge branch 'development' into automate-icingaweb2-setup

arjanoosting committed:
Specify default more clearly

arjanoosting committed:
Merge branch 'development' into automate-icingaweb2-setup

arjanoosting committed:
Support install icingaweb on additional hosts

arjanoosting committed:
handle lists

arjanoosting committed:
Use complete path to binary

arjanoosting committed:
Merge branch 'development' into automate-icingaweb2-setup

arjanoosting committed:
Only do set_fact if the hostvar is not set yet or incorrect and add an
option (i2_bootstrap_cacheable) to cache the values.
This will help with hosts which are temporarily down

arjanoosting committed:
Merge branch 'development' into automate-icingaweb2-setup

arjanoosting committed:
Remove space after comma in lists

arjanoosting committed:
Fix removal of unmanaged files which sometimes removed to much

arjanoosting committed:
Merge branch 'development' into automate-icingaweb2-setup

arjanoosting committed:
Remove when clause which causes configuration to be lost

arjanoosting committed:
Merge branch 'development' into automate-icingaweb2-setup

arjanoosting committed:
Handle hosts in the all group

arjanoosting committed:
Merge branch 'development' into automate-icingaweb2-setup

arjanoosting committed:
Another fix to handle hosts in the all group

arjanoosting committed:
Merge branch 'development' into automate-icingaweb2-setup

arjanoosting committed:
icinga_ido should use latin1.
See discussion https://community.icinga.com/t/umlauts-and-special-characters-in-icingaweb2/6415/6

arjanoosting committed:
Only install apt-transport-https when we are on an apt version which doesn't support https out of the box

arjanoosting committed:
Merge branch 'development' into automate-icingaweb2-setup
